src/base/result_decoder.re

```diff
@@ -575,6 +575,43 @@ let rec unify_document_schema = (config, document) => {
       ...rest,
     ] => [
       {
+        let with_decoder =
+          switch (fg_directives |> find_directive("bsDecoder")) {
+          | None => Ok(None)
+          | Some({item: {d_arguments, _}, span}) =>
+            switch (find_argument("fn", d_arguments)) {
+            | None =>
+              Error(
+                make_error(
+                  error_marker,
+                  config.map_loc,
+                  span,
+                  "bsDecoder must be given 'fn' argument",
+                ),
+              )
+            | Some((_, {item: Iv_string(fn_name), span})) =>
+              Ok(
+                Some(
+                  structure =>
+                    Res_custom_decoder(
+                      config.map_loc(span),
+                      fn_name,
+                      structure,
+                    ),
+                ),
+              )
+            | Some((_, {span, _})) =>
+              Error(
+                make_error(
+                  error_marker,
+                  config.map_loc,
+                  span,
+                  "The 'fn' argument must be a string",
+                ),
+              )
+            }
+          };
```

Comment:
There should in principle be no problems abstracting this out and using it to wrap fields, fragments _and_ mutations.

Reply:
I'm ok with leaving it as is. A lot will change with #67 and we'll have to do major code cleanup after that.

Reply:
Got it.